ERCS/erc-7794.md

```diff
@@ -0,0 +1,820 @@
+---
+eip: 7794
+title: Grant Registry
+description: A cross-network registry contract to manage and track grants, enhancing transparency and  interoperability for grant programs.
+author: Guilherme Neves (@0xneves)
+discussions-to: https://ethereum-magicians.org/t/erc-7794-grant-registry/20791
+status: Draft
+type: Standards Track
+category: ERC
+created: 2024-10-22
+---
+
+## Abstract
+
+This proposal introduces a Grant Registry contract designed to standardize the registration, management, and tracking of grants across multiple blockchains. The contract organizes grant data into distinct categories, separating immutable fields such as chain ID and community from mutable fields like milestones and participants. It allows for modular handling of disbursements and the addition of external links for off-chain data. By emitting key events throughout the grant lifecycle, the registry enables external protocols to fetch data efficiently, promoting transparency, interoperability, and better insights into the performance of grant programs.
+
+## Motivation
+
+The Ethereum ecosystem currently lacks a standardized way to manage and track grants across different chains and programs, leading to inefficiencies and fragmentation. Each grant program has its own distinct interface, processes, and management mechanisms, which creates barriers for both funders and grantees. These issues hinder transparency, complicate the tracking of fund disbursements, and make it difficult to evaluate the overall effectiveness of grant programs across different networks.
+
+The lack of interoperability between grant programs further exacerbates the problem, as projects and contributors often work across multiple blockchains. This makes it challenging to aggregate data, monitor milestones, and assess grantee performance in a consistent manner.
+
+The Grant Registry contract solves these issues by introducing a unified standard that ensures all grants can be registered, tracked, and managed consistently, regardless of the underlying chain or program. This approach not only simplifies the lifecycle management of grants but also fosters better collaboration between communities, allowing for more competitiveness and better tracking of progress. Additionally, the standardization of data opens the door for more insightful analytics, enabling protocols to measure the impact of grants in a much more streamlined and transparent way.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+### **Contract Interface**
+
+```solidity
+// SPDX-License-Identifier: MIT
```

Comment:
Code inline in the document must be `CC0-1.0` licensed. If you'd like to keep the code `MIT`, you can move it to your `assets/...` directory and link to it there.